cypress-io/cypress
# chore: wire up Cypress Studio

- Closes https://github.com/cypress-io/cypress/issues/23336

**Note** targeting a feature branch, `feat/studio-cypress-10`.

### User facing changelog


N/A - internal for now

### Additional details


Wire up Cypress Studio in Cypress 10. There's a bit going on, and it's not fully complete but works at a basic level. It's going into a feature branch, so things like tests, some cleanup etc will come before we merge the whole thing into develop.

Learn about Studio: https://docs.cypress.io/guides/references/cypress-studio#What-you-ll-learn

### Steps to test


WIP, there's probably lots of edge cases that are kind of busted so I wouldn't go too far. Here's a demo of how it works, you are welcome to play around and try it out!

The final review will be towards the end once we've got a proper UI in place. It's hard to really review without the full Studio UI in place - that's why it's targeting a feature branch.

Here's a quick demo (not sure why the test fails, but the wiring is working - you can see it generated a spec from my actions).

https://user-images.githubusercontent.com/19196536/185599079-8505f97c-a7fd-45ef-ab18-fccc9a500754.mp4

Any small bugs or minor fixes will be done in https://github.com/cypress-io/cypress/issues/23461, which is the feature branch. The final review of this is where we need to be more thorough and deliberate.

### How has the user experience changed?


### PR Tasks
